template/shared.tmpl

```diff
@@ -359,6 +359,71 @@
                 {{end}}
             {{end}}
 
+            {{if eq $property.Type "map[string]string"}}
+                {{if eq $property.ID "X-QS-MetaData" }}
+                    XQSMetaDataIsValid := true
+                    wrongKey := ""
+                    wrongValue := ""
+                        if *v.XQSMetaData!= nil {
+                            metadataValuelength := 0
+                            metadataKeylength := 0
+
+                            for k,v := range *v.XQSMetaData {
```

Comment:
Maybe we can add a util func like `IsMetaDataValid` in `utils` and call it in template which could be better understanding and testing.

Reply:
Ignore this, I'm planning a refactor for template.